rct229/rulesets/ashrae9012019/data_fns/table_7_8_fns.py

```diff
@@ -0,0 +1,113 @@
+from pint import Quantity
+
+from rct229.schema.config import ureg
+from rct229.rulesets.ashrae9012019.data import data
+from rct229.rulesets.ashrae9012019.data_fns.table_utils import find_osstd_table_entries
+from rct229.utils.assertions import assert_
+
+
+def table_7_8_lookup(
+    equipment_type: str, input_power: Quantity, draw_pattern: str = ""
+) -> list:
+    """Returns the service water heater efficiency value(s) and metrics for a baseline service water heater as
+    required by ASHRAE 90.1 Table 7.8
+
+    Parameters
+    ----------
+    equipment_type : str
+        One of "Electric storage water heater" or "Gas storage water heater"
+    input_power : Quantity
+        The input power of the service water heater, in kW for Electric and in Btu/h for Gas
+    draw_pattern : str
+        One of: "Very small", "Low", "Medium", "High", or "" if not applicable
+
+    Returns
+    -------
+    list of dict
+        [
+            {
+                "efficiency_value": 0.80,
+                "efficiency_metric": "UNIFORM_EFFICIENCY_FACTOR",
+            },
+            {
+                "efficiency_value": 0.05,
+                "efficiency_metric": "STAND_BY_LOSS_FRACTION",
+            },
+        ]
+
+    """
+    # Validate equipment type
+    valid_equipment_types = [
+        "Electric storage water heater",
+        "Gas storage water heater",
+    ]
+    assert_(
+        equipment_type in valid_equipment_types,
+        f"Invalid equipment type. Must be one of {valid_equipment_types}",
+    )
```

Comment:
I think maybe just use `assert` instead of `assert_`? I check this again and feel like this will only be triggered by software implementation error instead of project file.
@JacksonJ-KC thoughts?